src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { st, classes } from './Dialog.st.css';
+import { DATA_HOOKS } from './constants';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+
+import { Modal } from '../Modal';
+import { IconButton } from '../IconButton';
+import { Close as CloseIcon } from '../../assets/icons';
+import { TPAComponentProps } from '../../types';
+import { KEY_CODES } from '../../common/keyCodes';
+
+export interface DialogProps extends TPAComponentProps {
+  /** Whether the modal is opened */
+  isOpen?: boolean;
+  /** Whether to use our focus trap or to manage the focus manually. */
+  manualFocus?: boolean;
+  /** Adds a ref for the close button. This prop might be used when you choose the manual focus. */
+  closeButtonRef?: React.RefObject<HTMLButtonElement>;
+  /** Callback for when the close button is clicked */
+  onClose?(): void;
+  /** Children to render inside dialog */
+  children?: React.ReactNode;
+  /** Defines a string value that labels the dialog element. Optional. */
+  'aria-label'?: string;
+  /** Identifies the element that labels the dialog element. Optional. */
+  'aria-labelledby'?: string;
+  /** Gives the dialog an accessible description by referring to the dialog content that describes the primary message or purpose of the dialog. Optional. */
+  'aria-describedby'?: string;
+  /** Defines a string value that labels the close button element. Optional. */
+  'close-button-aria-label'?: string;
+  /** Identifies the element that labels the close button element. Optional. */
+  'close-button-aria-labelledby'?: string;
+}
+
+interface DefaultProps {
+  isOpen: boolean;
+  manualFocus: boolean;
+}
+
+/** Dialog */
+export class Dialog extends React.Component<DialogProps> {
+  static displayName = 'Dialog';
+  static defaultProps: DefaultProps = {
+    isOpen: false,
+    manualFocus: false,
+  };
+
+  _addEscEventListener() {
+    if (this.props.isOpen) {
+      document.addEventListener('keyup', this._onEscKeyup);
+    }
+  }
+
+  _removeEscEventListener() {
+    document.removeEventListener('keyup', this._onEscKeyup);
+  }
```

Comment:
I'm thinking maybe this logic should be in the Modal, no? wdyt?
Also, I guess we'll need this to **not** work in mobile.
In these cases, you can use the context like this:
```jsx
class MyClass extends React.Component {
  static contextType = MyContext;
  render() {
    let value = this.context;
    /* render something based on the value */
  }
}
```

Reply:
yes, maybe I should move it for the modal.
But then should it added only when "closeOnClickOutside" is true? This is the reason that I put it in the dialog cause it seems like an alternative for the X button.
WDYT?

Reply:
🤔  good question
I guess the safest thing to do is add a `closeOnEsc` prop.
How do WSR modal's work? do they always close on esc?

Reply:
Yes sounds like a good idea.
In WSR the Modal uses ReactModal which has a prop "shouldCloseOnEsc" that by default true.
I think I will do the same in our Modal :)

Reply:
Hmmm, the only thing is that usually it's best not to have `true` as a default value.
So maybe we should use a prop like `disableEsc` or `withoutEsc`?
wdyt?